[GCI-258]: Add relevant icons to OpenMRS AddOns modules

f4ww4z opened the pull request:
https://issues.openmrs.org/browse/GCI-258

I added corresponding icons to existing addons. All Refapp modules except Legacy UI and OWA module should have an icon now.

f4ww4z commented on src/main/resources/add-ons-to-index.json:
```diff
@@ -2593,10 +2621,10 @@
       "tags": [
         "reporting"
       ],
-      "icon": "bar-chart",
```
I just swapped this to follow the convention as described [here](https://github.com/openmrs/openmrs-contrib-addonindex/blob/da43a6c5f44eadad8c96866f3f193fe38655e758/src/main/resources/add-ons-to-index.json#L1244).

reubenvarghese1 commented on src/main/resources/add-ons-to-index.json:
```diff
@@ -4023,6 +4073,7 @@
       "uid": "org.openmrs.module.allergy-ui-module",
       "type": "OMOD",
       "name": "Allergy UI Module",
+      "icon": "user-circle-o",
```
Using this icon for this module doesn't make sense. A user icon is a rather common icon which gives us no info about the utility of the module

reubenvarghese1 commented on src/main/resources/add-ons-to-index.json:
```diff
@@ -3931,6 +3978,7 @@
       "uid": "org.openmrs.module.openmrs-fhir-module",
       "type": "OMOD",
       "name": "OpenMRS FHIR Module",
+      "icon": "list-ol",
```
Doesn't make much sense to add this here

f4ww4z commented:
I replaced the icons you wanted me to change with better icons @reubenvarghese1 . Are they good icons to use?

reubenvarghese1 commented on src/main/resources/add-ons-to-index.json:
```diff
@@ -697,6 +704,7 @@
       "uid": "org.openmrs.module.dataentrystatistics",
       "type": "OMOD",
       "name": "Data Entry Statistics",
+      "icon": "pie-chart",
```
I suggest we use bar-chart or something of that sort for this one

reubenvarghese1 commented on src/main/resources/add-ons-to-index.json:
```diff
@@ -849,6 +858,7 @@
       "uid": "org.openmrs.module.emrapi",
       "type": "OMOD",
       "name": "EMR API Module",
```
This one isn't really relevant so I guess this should be replaced/ deleted

reubenvarghese1 commented on src/main/resources/add-ons-to-index.json:
```diff
@@ -849,6 +858,7 @@
       "uid": "org.openmrs.module.emrapi",
       "type": "OMOD",
       "name": "EMR API Module",
+      "icon": "hospital-o",
```
This one isn't really relevant so I guess this should be replaced/ deleted

reubenvarghese1 commented on src/main/resources/add-ons-to-index.json:
```diff
@@ -1083,6 +1096,7 @@
       "uid": "org.openmrs.module.formentry",
       "type": "OMOD",
       "name": "Form Entry",
+      "icon": "file-code-o",
```
This one isn't really relevant so I guess this should be replaced/ deleted

reubenvarghese1 commented on src/main/resources/add-ons-to-index.json:
```diff
@@ -1349,6 +1364,7 @@
       "status": "DEPRECATED",
       "type": "OMOD",
       "name": "HTML Form Entry Extensions for OpenMRS 1.9 Module",
+      "icon": "code",
```
This one isn't really relevant so I guess this should be replaced/ deleted

reubenvarghese1 commented on src/main/resources/add-ons-to-index.json:
```diff
@@ -1396,6 +1413,7 @@
       "uid": "org.openmrs.module.htmlformflowsheet",
       "type": "OMOD",
       "name": "HTML Form Flowsheet",
+      "icon": "file-code-o",
```
This one isn't really relevant so I guess this should be replaced/ deleted

reubenvarghese1 commented on src/main/resources/add-ons-to-index.json:
```diff
@@ -1421,6 +1439,7 @@
       "uid": "org.openmrs.module.htmlwidgets",
       "type": "OMOD",
       "name": "HTML Widgets",
+      "icon": "html5",
```
Not everyone will recognize the logo. Hence, it might seem as though this icon is not matching the module name and hence I feel this must be deleted

reubenvarghese1 commented on src/main/resources/add-ons-to-index.json:
```diff
@@ -1547,6 +1566,7 @@
       "uid": "org.openmrs.module.jasperreport",
       "type": "OMOD",
       "name": "Jasper Report",
+      "icon": "file-o",
```
Maybe use file-text-o ?

reubenvarghese1 commented on src/main/resources/add-ons-to-index.json:
```diff
@@ -1830,6 +1851,7 @@
       "uid": "org.openmrs.module.metadatadeploy",
       "type": "OMOD",
       "name": "Metadata Deploy",
+      "icon": "plus-square-o",
```
This one isn't really relevant so I guess this should be replaced/ deleted

reubenvarghese1 commented on src/main/resources/add-ons-to-index.json:
```diff
@@ -2148,6 +2173,7 @@
       "uid": "org.openmrs.module.openhmis-backboneforms",
       "type": "OMOD",
       "name": "OpenHMIS Backbone Forms",
+      "icon": "hospital-o",
```
This one isn't really relevant so I guess this should be replaced/ deleted

reubenvarghese1 commented on src/main/resources/add-ons-to-index.json:
```diff
@@ -2276,6 +2302,7 @@
       "uid": "org.openmrs.module.patientmatching",
       "type": "OMOD",
       "name": "Patient Matching",
+      "icon": "handshake-o",
```
This one isn't really relevant so I guess this should be replaced/ deleted. The link icon makes more sense here. To understand why, check out the wiki of the module here https://wiki.openmrs.org/display/docs/Patient+Matching+Module

reubenvarghese1 commented on src/main/resources/add-ons-to-index.json:
```diff
@@ -2455,6 +2482,7 @@
       "uid": "org.openmrs.module.providermanagement",
       "type": "OMOD",
       "name": "Provider Management Module",
+      "icon": "user-md",
```
This one isn't really relevant so I guess this should be replaced/ deleted

reubenvarghese1 commented on src/main/resources/add-ons-to-index.json:
```diff
@@ -2625,6 +2653,7 @@
       "uid": "org.openmrs.module.reportingcompatibility",
       "type": "OMOD",
       "name": "Reporting Compatibility",
+      "icon": "share-square-o",
```
This one isn't really relevant so I guess this should be replaced/ deleted

reubenvarghese1 commented on src/main/resources/add-ons-to-index.json:
```diff
@@ -2659,6 +2688,7 @@
       "uid": "org.openmrs.module.reportingrest",
       "type": "OMOD",
       "name": "reportingrest",
+      "icon": "bars",
```
This one isn't really relevant so I guess this should be replaced/ deleted

reubenvarghese1 commented on src/main/resources/add-ons-to-index.json:
```diff
@@ -2733,6 +2763,7 @@
       "uid": "org.openmrs.module.researchencounters",
       "type": "OMOD",
       "name": "Research Encounters",
+      "icon": "file-text-o",
```
This one isn't really relevant so I guess this should be replaced/ deleted

reubenvarghese1 commented on src/main/resources/add-ons-to-index.json:
```diff
@@ -2871,6 +2902,7 @@
       "status": "INACTIVE",
       "type": "OMOD",
       "name": "Quartz Scheduler",
+      "icon": "calendar",
```
This one isn't really relevant so I guess this should be replaced/ deleted.Check https://wiki.openmrs.org/display/docs/Quartz+Scheduler+Module for more info

reubenvarghese1 commented on src/main/resources/add-ons-to-index.json:
```diff
@@ -3225,6 +3260,7 @@
       "uid": "org.openmrs.module.usagestatistics",
       "type": "OMOD",
       "name": "Usage Statistics",
+      "icon": "line-chart",
```
COuld use bar-chart probably

reubenvarghese1 commented on src/main/resources/add-ons-to-index.json:
```diff
@@ -3535,6 +3573,7 @@
       "uid": "org.openmrs.module.form-entry-app-module",
       "type": "OMOD",
       "name": "Form Entry App Module",
+      "icon": "file-text",
```
This one isn't really relevant so I guess this should be replaced/ deleted.

reubenvarghese1 commented on src/main/resources/add-ons-to-index.json:
```diff
@@ -3602,6 +3641,7 @@
       "uid": "org.openmrs.module.reference-metadata-module",
       "type": "OMOD",
       "name": "Reference Metadata Module",
+      "icon": "link",
```
This one isn't really relevant so I guess this should be replaced/ deleted.

reubenvarghese1 commented on src/main/resources/add-ons-to-index.json:
```diff
@@ -3627,6 +3667,7 @@
       "uid": "org.openmrs.module.core-apps-module",
       "type": "OMOD",
       "name": "Core Apps Module",
+      "icon": "code-fork",
```
This one isn't really relevant so I guess this should be replaced/ deleted.

reubenvarghese1 commented on src/main/resources/add-ons-to-index.json:
```diff
@@ -3655,6 +3696,7 @@
       "uid": "org.openmrs.module.reference-demo-data-module",
       "type": "OMOD",
       "name": "Reference Demo Data Module",
+      "icon": "clone",
```
This one isn't really relevant so I guess this should be replaced/ deleted.

reubenvarghese1 commented on src/main/resources/add-ons-to-index.json:
```diff
@@ -3852,6 +3897,7 @@
       "uid": "org.openmrs.module.registration-core-module",
       "type": "OMOD",
       "name": "Registration Core Module",
+      "icon": "users",
```
This one isn't really relevant so I guess this should be replaced/ deleted. User-plus would fit better

reubenvarghese1 commented on src/main/resources/add-ons-to-index.json:
```diff
@@ -3931,6 +3978,7 @@
       "uid": "org.openmrs.module.openmrs-fhir-module",
       "type": "OMOD",
       "name": "OpenMRS FHIR Module",
+      "icon": "list-alt",
```
This one isn't really relevant so I guess this should be replaced/ deleted.

reubenvarghese1 commented on src/main/resources/add-ons-to-index.json:
```diff
@@ -4023,6 +4073,7 @@
       "uid": "org.openmrs.module.allergy-ui-module",
       "type": "OMOD",
       "name": "Allergy UI Module",
+      "icon": "hand-paper-o",
```
This one isn't really relevant so I guess this should be replaced/ deleted.

reubenvarghese1 commented:
Looks like quite a modules have been addressed by both  you and @tomaszmueller ; So, to fix the merge conflict. I will review each of your PRs individually as though the other doesn't exist yet. Once my review is complete. The one who submits a "merge-ready" PR first will get his PR merged while the other will just have to delete the ones already addressed by the merged commit and submit his existing unique changes. Hope this sounds fair.

f4ww4z commented:
My changes:

- Replaced org.openmrs.module.dataentrystatistics icon from 'pie-chart' to 'bar-chart'
- Removed org.openmrs.module.emrapi 'hospital-o' icon
- Replaced org.openmrs.module.formentry icon 'file-code-o' to 'pencil-square-o'
- Replaced org.openmrs.module.htmlformflowsheet icon from 'file-code-o' to 'code'
- Removed org.openmrs.module.htmlwidgets 'html5' icon
- Replaced org.openmrs.module.jasperreport icon from 'file-o' to 'file-text-o'
- Removed org.openmrs.module.metadatadeploy 'plus-square-o' icon
- Replaced org.openmrs.module.openhmis-backboneforms icon from 'hospital-o' to 'book'
- Replaced org.openmrs.module.patientmatching icon from 'handshake-o' to 'link'
- Replaced org.openmrs.module.providermanagement icon from 'user-md' to 'briefcase'
- Replaced org.openmrs.module.reportingcompatibility icon from 'share-square-o' to 'bar-chart'
- Replaced org.openmrs.module.reportingrest icon from 'bars' to 'bar-chart'
- Replaced org.openmrs.module.researchencounters icon from 'file-text-o' to 'files-o'
- Replaced org.openmrs.module.schedulerquartz icon from 'calendar' to 'tasks'
- Replaced org.openmrs.module.usagestatistics icon from 'line-chart' to 'bar-chart'
- Replaced org.openmrs.module.form-entry-app-module icon from 'file-text' to 'pencil-square-o'
- Replaced org.openmrs.module.reference-metadata-module icon from 'link' to 'database'
- Replaced org.openmrs.module.core-apps-module icon from 'code-fork' to 'th-large'
- Replaced org.openmrs.module.reference-demo-data-module icon from 'clone' to 'laptop'
- Replaced org.openmrs.module.registration-core-module icon from 'users' to 'user-plus'
- Removed org.openmrs.module.openmrs-fhir-module 'list-alt' icon
- Removed org.openmrs.module.allergy-ui-module 'hand-paper-o' icon

I know I'll have to resolve the conflicts with #33 so I'll do that after the @tomekdo3 's PR is merged.
